components/dataHub/adminService.c

```diff
@@ -680,6 +680,18 @@ void admin_DeleteObs
 )
 //--------------------------------------------------------------------------------------------------
 {
+    // Verify that the path is within the /obs namespace, and if it's an absolute path, convert
+    // it into a relative path.
+    if (strncmp(path, "/obs/", 5) == 0)
+    {
+        path += 5;
+    }
+    else if (path[0] == '/')
+    {
+        LE_ERROR("Path is outside the /obs/ namespace (%s).", path);
+        return;
+    }
+
```

Comment:
I prefer to avoid hardcoding string lengths.  It's easy for this kind of thing to introduce bugs later when someone decides that "/obs/" should be changed to "/observations/".  I also prefer to move error condition checking as early as possible and the elimination of the else case because I think it improves readability slightly.  This is all just nit-picky stuff.  The code as-is looks correct.
```
// Absolute paths must be withing the /obs namespace and are converted to a relative path
if (path[0] == '/')
{
    const char *obsPrefix = "obs/";
    if (strncmp(&path[1], obsPrefix, strlen(obsPrefix)) != 0)
    {
        LE_ERROR("Path is outside the /obs/ namespace (%s).", path);
        return;
    }
    path += strlen(obsPrefix) + 1;
}
```

Reply:
I don't care about this enough to hold this pull request up any longer.